### PR TITLE
Add QrCode & Copy to ShapeShift

### PR DESCRIPTION
**Pull request by jacogr**
Closes https://github.com/ethcore/parity/issues/4238

QrCode output verified with the BlockChain.info app, able to read & translate.

parity 2017-01-27 10-10-04

### CODE REVIEW

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js [hunk 1]
```diff
@@ -61,9 +63,7 @@ export default class AwaitingDepositStep extends Component {
             } }
           />
         </div>
-        <div className={ styles.hero }>
```
Comment (ngotchac) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js]: Is the class `.hero` still used anywhere ?

**Comment by ngotchac** [comment]
Same comment as for #4329 : don't we want to use `bitcoin:...` URLs ?

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js (hunk 1 above)
Comment (jacogr) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js]: Yeap. Both in AwaitingExchange & CompletedStep.

**Comment by jacogr** [comment]
Could make it clickable with `bitcoin:`, good point.

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js (hunk 1 above)
Comment (ngotchac) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js]: Ah right, wasn't looking in the right folder. My bad !

**Comment by jacogr** [comment]
Added `bitcoin:` clicky (nice, it works with my exchange, makes my life easier) and made the default size smaller.

parity 2017-01-27 13-19-09

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js [hunk 2]
```diff
@@ -79,4 +79,24 @@ export default class AwaitingDepositStep extends Component {
       </div>
     );
   }
+
+  renderAddress (depositAddress) {
+    return (
+      <div className={ styles.addressInfo }>
+        <a
+          href={ `bitcoin:${depositAddress}` }
```
Comment (ngotchac) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js]: Shouldn't this be only for BTC transfers ?

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js [hunk 3]
```diff
@@ -79,4 +79,24 @@ export default class AwaitingDepositStep extends Component {
...
+          target='_blank'
+        >
+          <QrCode
+            className={ styles.qrcode }
+            value={ depositAddress }
```
Comment (ngotchac) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js]: Don't we want to use the `bitcoin:${despositAddress}` format for BTC exchanges ? It feels unbalanced, but aren't QR code scanners on mobile suppose to trigger BTC app if available ?

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js (hunk 2 above)
Comment (jacogr) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.js]: &^%$! So excited to have it working that I didn't pay attention.

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.spec.js [hunk 4]
```diff
@@ -30,11 +33,12 @@ function render () {
       } }
     />
   );
+  instance = component.instance();
 
   return component;
 }
 
-describe('modals/Shapeshift/AwaitingDepositStep', () => {
+describe.only('modals/Shapeshift/AwaitingDepositStep', () => {
```
Comment (ngotchac) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.spec.js]: Might want to get rid of this one ;)

File: js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.spec.js [hunk 5]
```diff
@@ -48,4 +52,61 @@ describe('modals/Shapeshift/AwaitingDepositStep', () => {
...
+
+        it('renders the QrCode', () => {
+          expect(qr.length).to.equal(1);
+        });
+
+        it('passed the address', () => {
+          expect(qr.props().value).to.equal(TEST_ADDRESS);
+        });
+
+        describe('protocol link', () => {
```
Comment (ngotchac) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.spec.js]: Is under `QrCode` test the right place for these tests ?

Comment (jacogr) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.spec.js]: The link wraps the QrCode. Not attached to it either way, could be outside as well.

Comment (ngotchac) [inline js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.spec.js]: Ok

**Pull request merged**